scripts/reg_accesses/reg_display.py

```diff
@@ -0,0 +1,126 @@
+
+    plt.bar(range(l_size), counters)
+    plt.xticks(range(l_size), keys)
+    if len(counters):
+        plt.yticks(range(counters[0] + 1))
+    else: # Case for empty lists
+        plt.yticks([0, 1])
+    
+    if name == "rs":
+        plt.title("Registers used most frequently as source registers")
+        plt.xlabel("Registers")
+        plt.ylabel("Number of times accessed")
+    elif name == "rd":
+        plt.title("Registers used most frequently as destination registers")
+        plt.xlabel("Registers")
+        plt.ylabel("Number of times read to")
+    elif name == "imm":
+        plt.title("Most frequent immediate values")
+        plt.xlabel("Immediates")
+        plt.ylabel("Number of times used")
+    elif name == "address_offset":
+        plt.title("Most frequent address offset values")
+        plt.xlabel("Immediates (Offsets)")
+        plt.ylabel("Number of times used")
+    elif name == "branch_offset":
+        plt.title("Most frequent branch offsets values")
+        plt.xlabel("Immediates (Branch offsets)")
+        plt.ylabel("Number of times used")
+    elif name == "shift":
+        plt.title("Most frequent shift values")
+        plt.xlabel("Immediates (Shifts)")
+        plt.ylabel("Number of times used")
+    elif name == "arith":
+        plt.title("Most frequent arithmetic values")
+        plt.xlabel("Immediates (Arithmetic values)")
+        plt.ylabel("Number of times used")
+    
+    plt.show()
+
+# Function used to see how much each register available in the system was used as a source or destination register
+def reg_distribution_plot(all_regs):
+    reg_names = [x[0] for x in all_regs]
+    rs_list = [x[1]["rs"] for x in all_regs]
+    rd_list = [x[1]["rd"] for x in all_regs]
+
+    plt.bar(range(len(reg_names)), height=rs_list, width=0.4, align='edge', label="rs")
+    plt.bar(range(len(reg_names)), height=rd_list, width=-0.4, align='edge', label="rd")
+
+    plt.xticks(range(len(reg_names)), reg_names)
+    plt.xlabel("Registers")
+    plt.ylabel("Frequency")
+    plt.legend()
+
+    plt.show()
+
+# Visualise the distribution of the immediates and what they are used for
+def imm_distribution_plot(off, branch_off, shift, arith):
+    sums_list = [0]*4
+    sums_list[0] = sum(i for _,i in off)
+    sums_list[1] = sum(i for _,i in branch_off)
+    sums_list[2] = sum(i for _,i in shift)
+    sums_list[3] = sum(i for _,i in arith)
+    sums_list = sorted(sums_list, reverse=True)
+
+    plt.bar(range(len(sums_list)), sums_list)
+    plt.xticks(range(len(sums_list)), ["Immediate total", "Address offsets", 
+        "Branch offsets", "Shift", "Arithmetic"])
+    plt.title("Immediate distribution of "+str(sum(sums_list))+" immediates")
+    plt.xlabel("Immediate type")
+    plt.ylabel("Frequency")
+    plt.show()
+
+def main():
+    # Read in the associated lists and store in the data dictionary
+    with open('scripts/reg_accesses/test-rs.txt', 'r') as dump:
```

Comment:
Sorry, was just testing out using JSON files and I'm planning on having it properly integrated when I add these scripts to the Makefile. I plan on having an input argument in the python script that determines the file path for the JSON file and so this will be determined in the Makefile

Reply:
This should have now been resolved in the latest commit.